app/code/Magento/ImportExport/Model/Import/AbstractEntity.php

```diff
@@ -831,7 +831,7 @@ public function validateData()
                 if (!$this->isAttributeParticular($columnName)) {
                     if (trim($columnName) == '') {
                         $emptyHeaderColumns[] = $columnNumber;
-                    } elseif (!preg_match('/^[a-z][a-z0-9_]*$/', $columnName)) {
+                    } elseif (!preg_match('/^[a-z][\w]*$/u', $columnName)) {
```

Comment:
Why first character is supposed to be a Latin letter?

Reply:
I'd say that a regex like `'/^[\w]+$/u'` would work perfectly fine as well, but I wanted to keep old behavior for whatever reason it was designed this way. Should I change it to `'/^[\w]+$/u'`?

Reply:
Yes. So that name like `µg_vitamin_a_µg` is also supported. Not sure it can start from digit or underscore though, could you check attribute validation when you create/update it via Admin UI please?

Reply:
Sure thing. I'll get back to you.

Reply:
The attribute code validation when saving happens right here: [Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php#L161](https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php#L161). The regex seems to be `^[a-z\x{600}-\x{6FF}][a-z\x{600}-\x{6FF}_0-9]{0,30}$`. 

It starts to look like we need a validation class for attribute codes in general, so we can unify this stuff. But that's a story for another time. 

@orlangur What's your suggestion for now?

Reply:
@tdgroot ok, so it's not "Magento_ImportExport not supporting unicode characters in column names" actually but such attribute code are generally forbidden. I've raised an internal discussion to understand the correct approach.

Reply:
@orlangur sure, I'd like to add something to it. I created the attribute in my example using the Magento API methods. To me it seems that validation of the attribute code shouldn't happen in a controller, but rather in the EAV model/repository class.